Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h

```diff
@@ -107,7 +107,7 @@ class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
   itkSetObjectMacro(TriangleMesh, InputMeshType);
 
   /** Get Gauss curvature. */
-  itkGetModifiableObjectMacro(GaussCurvatureData, DoubleVectorContainer);
+  itkGetConstObjectMacro(GaussCurvatureData, DoubleVectorContainer);
```

Comment:
If it all works fine with just a `itkGetConstObjectMacro` call here, it's fine to me as well. Just want to mention that `itkGetModifiableObjectMacro` defines _both_ a "modifiable" and a `const` getter, even when `ITK_FUTURE_LEGACY_REMOVE` is defined:

https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc02/Modules/Core/Common/include/itkMacro.h#L1131-L1133 

So strictly speaking, you should still be able to have "const" access, by calling `GetGaussCurvatureData()`, even when `itkGetModifiableObjectMacro`  is being used. (Please correct me if I'm wrong.)

Having said so, `itkGetConstObjectMacro` is fine to me as well.

Reply:
Yes, you are right. Both the access methods will be defined. But in the master code in the first Get call, we have called GetModifiableGaussCurvatureData and then later called GetGaussCurvatureData. And the compiler is throwing error because the pointer returned is of different types.

```
candidate function not viable: no known conversion from 'const
itk::TriangleMeshCurvatureCalculator<itk::Mesh<double, 3>>::DoubleVectorContainer *' 
(aka 'const VectorContainer<unsigned long, double> *') to 
'itk::SmartPointer<itk::VectorContainer<unsigned long, double>>' for 
1st argument operator=(SmartPointer r) noexcept
```

But we only wanted to give read access to the curvature data so itkGetConstObjectMacro is the right choice.

Reply:
@PranjalSahu I think I agree with you, but just to be sure: Is there a reasonable use case, retrieving a pointer to the curvature data from the calculator, and then wanting to modify that data "in-place" (rather than making a copy)?  

If such a use case doesn't seem to be there at the moment, your PR is fine to me.  As long as it passes all CI checks  😃